Optimizer/FcOptimizer.hs

```diff
@@ -0,0 +1,10 @@
+module Optimizer.FcOptimizer where
+
+import Utils.AssocList
+
+-- | Data constructor environment, listing type variables for type and data constructors
+type DCEnv = (AssocList FcTyCon FcTyConInfo, AssocList FcDataCon FcDataConInfo)
+
+type FcOptM = UniqueSupplyT 
+
+type FcOptPass = DCEnv -> UniqueSupply -> FcOptProgram -> 
```

Comment:
Something seems to be missing here.
Also, it might be helpful to already write the main function signature here.

Reply:
yeah this file should not have been included into the commit yet...